Investigate intermittent test failure in dropwizard-client-poller

Recently we saw this test failure:

```
 [ERROR]   DefaultClientPollerStatisticsTest.testIncrementCount_SetsLastAttemptTime(SoftAssertions) 
    Multiple Failures (1 failure)
    -- failure 1 --
    Expecting actual:
      1650674759819L
    to be greater than:
      1650674760096L
    
    at DefaultClientPollerStatisticsTest.justOccurred(DefaultClientPollerStatisticsTest.java:82)
 ```

Full error:

```
   [INFO] Running org.kiwiproject.dropwizard.poller.metrics.DefaultClientPollerStatisticsTest
    20:46:04.622 [main] WARN org.kiwiproject.dropwizard.poller.metrics.DefaultClientPollerStatistics - Skipping measurement 200 because lock was not obtained in 1 MILLISECONDS
    20:46:04.632 [main] WARN org.kiwiproject.dropwizard.poller.metrics.DefaultClientPollerStatistics - Skipping measurement 100 because lock was not obtained in 1 MILLISECONDS
    [ERROR] Tests run: 16, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 10.155 s <<< FAILURE! - in org.kiwiproject.dropwizard.poller.metrics.DefaultClientPollerStatisticsTest
    [ERROR] org.kiwiproject.dropwizard.poller.metrics.DefaultClientPollerStatisticsTest.testIncrementCount_SetsLastAttemptTime(SoftAssertions)  Time elapsed: 2.25 s  <<< FAILURE!
    org.assertj.core.error.AssertJMultipleFailuresError: 
    
    Multiple Failures (1 failure)
    -- failure 1 --
    Expecting actual:
      1650674759819L
    to be greater than:
      1650674760096L
    
    at DefaultClientPollerStatisticsTest.justOccurred(DefaultClientPollerStatisticsTest.java:82)
    	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
    	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
    	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
    	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeAfterTestExecutionCallbacks$9(TestMethodTestDescriptor.java:233)
    	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeAllAfterMethodsOrCallbacks$13(TestMethodTestDescriptor.java:273)
    	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
    	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeAllAfterMethodsOrCallbacks$14(TestMethodTestDescriptor.java:273)
    	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
    	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeAllAfterMethodsOrCallbacks(TestMethodTestDescriptor.java:272)
    	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeAfterTestExecutionCallbacks(TestMethodTestDescriptor.java:232)
    	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:137)
    	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:66)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
    	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
    	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
    	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
    	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
    	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
    	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
    	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
    	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
    	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
    	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
    	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
    	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
    	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
    	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
    	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
    	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
    	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
    	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
    	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
    	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
    	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
    	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
    	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
    	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
    	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
    	at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:55)
    	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:223)
    	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:175)
    	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:139)
    	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
    	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
    	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
    	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
```

--- a/src/test/java/org/kiwiproject/dropwizard/poller/metrics/DefaultClientPollerStatisticsTest.java
+++ b/src/test/java/org/kiwiproject/dropwizard/poller/metrics/DefaultClientPollerStatisticsTest.java
@@ -56,30 +56,31 @@ class DefaultClientPollerStatisticsTest {
     }
 
     @Test
-    void testIncrementCount_SetsLastAttemptTime(SoftAssertions softly) {
+    void testIncrementCount_SetsLastAttemptTime() {
         stats.incrementCount();
 
-        softly.assertThat(stats.totalCount()).isOne();
-        softly.assertThat(stats.lastAttemptTimeInMillis())
+        assertThat(stats.totalCount()).isOne();
+        assertThat(stats.lastAttemptTimeInMillis())
                 .isPresent()
-                .hasValueSatisfying(timeInMillis -> justOccurred(timeInMillis, softly));
+                .hasValueSatisfying(this::justOccurred);
     }
 
     @Test
-    void testIncrementSkipCount_SetsLastSkipTime(SoftAssertions softly) {
+    void testIncrementSkipCount_SetsLastSkipTime() {
         stats.incrementSkipCount();
 
-        softly.assertThat(stats.totalCount()).isZero();
-        softly.assertThat(stats.skipCount()).isOne();
-        softly.assertThat(stats.lastSkipTimeInMillis())
+        assertThat(stats.totalCount()).isZero();
+        assertThat(stats.skipCount()).isOne();
+        assertThat(stats.lastSkipTimeInMillis())
                 .isPresent()
-                .hasValueSatisfying(timeInMillis -> justOccurred(timeInMillis, softly));
+                .hasValueSatisfying(this::justOccurred);
     }
 
-    private void justOccurred(long timeInMillis, SoftAssertions softly) {
+    private void justOccurred(long timeInMillis) {
         long now = System.currentTimeMillis();
-        softly.assertThat(timeInMillis).isLessThanOrEqualTo(now);
-        softly.assertThat(timeInMillis).isGreaterThan(now - TimeUnit.SECONDS.toMillis(1));
+        assertThat(timeInMillis)
+                .isLessThanOrEqualTo(now)
+                .isGreaterThan(now - TimeUnit.SECONDS.toMillis(1));
     }
 
     @Test
@@ -160,9 +161,9 @@ class DefaultClientPollerStatisticsTest {
         List<Long> recentFailureTimes = stats.recentFailureTimesInMillis().collect(toList());
         Collections.reverse(recentFailureTimes);
 
-        //noinspection UnstableApiUsage
+        // noinspection UnstableApiUsage
         Streams.zip(failureDetails.stream(), recentFailureTimes.stream(),
-                (failureDetail, failureTime) -> Pair.of(failureDetail.get(DefaultClientPollerStatistics.FailedPollResult.TIME_KEY), failureTime))
+                        (failureDetail, failureTime) -> Pair.of(failureDetail.get(DefaultClientPollerStatistics.FailedPollResult.TIME_KEY), failureTime))
                 .forEach(pair ->
                         softly.assertThat(pair.getLeft())
                                 .describedAs("Failure detail time must match failure time")

